### PR TITLE
Updated code / tests for a full en-* run

### PR DESCRIPTION
**Pull request by rayborn**

### CODE REVIEW

**Comment by rayborn** [comment]
Duplicate of #193 doh

File: src/recipes/ios-promo/config.js [hunk 1]
```diff
@@ -21,13 +21,13 @@ let thankyou = "We hope that you enjoy Firefox on your mobile device!";
 let url      = "https://www.mozilla.org/en-US/firefox/mobile-download/";
 let button   = "Get it now";
 let branches = [
-  {
+  /*{
```
Comment (gregglind) [inline src/recipes/ios-promo/config.js]: so this is the fail branch?  put in a comment saying so.   // lowest performing branch

File: src/recipes/ios-promo/config.js [hunk 2]
```diff
@@ -49,7 +49,7 @@ module.exports = {
     branch: b,
     name: b.name,
     key: 'ios-promo',
-    version: 1,
+    version: 2,
```
Comment (gregglind) [inline src/recipes/ios-promo/config.js]: cool

File: src/recipes/ios-promo/config.js [hunk 3]
```diff
@@ -57,7 +57,7 @@ module.exports = {
     all: {
       sample: 10 * percent, // will be 1.0 when we launch
       restdays: null,  // irrelevant
-      locales: ["en-US", "en-GB"], // will be en-* locales when we launch
+      locales: ["en-US", "en-GB", "en-ZA"] // All English locales that we support
```
Comment (gregglind) [inline src/recipes/ios-promo/config.js]: en-ZA?  Truly!?  That's neat.  

As an alt solution, consider "en-*" and a regex test.

File: src/recipes/ios-promo/index.js [hunk 4]
```diff
@@ -108,13 +108,6 @@ let waitedEnough = function (restDays, last, now) {
 };
```
Comment (gregglind) [inline src/recipes/ios-promo/index.js]: I would probably leave it in.  It's dead code though.  I assume we are 10% everywhere?  That's our solution?

File: src/repairs.js [hunk 5]
```diff
@@ -16,5 +16,5 @@ let repairList = module.exports = [
   //require("./always"),
   require("./recipes/heartbeat-by-user-first-impression"),
   require("./recipes/pb-mode-survey"),
-  //require("./recipes/ios-promo")
+  require("./recipes/ios-promo")
```
Comment (gregglind) [inline src/repairs.js]: ok

File: src/recipes/ios-promo/index.js [hunk 6]
```diff
@@ -172,11 +162,6 @@ let shouldRun = function (userstate, config, extras) {
     return false;
   }
```
Comment (gregglind) [inline src/recipes/ios-promo/index.js]: ok

File: test/recipes/test-ios-promo.js [hunk 7]
```diff
@@ -124,9 +124,8 @@ describe("ios-promo", function () {
             let restdays = C.channels[channel].restdays;
             let now = Date.now();
             let ans = R.shouldRun(state, C.channels[channel],
-              {when: now - 1, lastRun: now - (restdays * days),
-               geoOK: true
-            })
+              {when: now - 1, lastRun: now - (restdays * days)}
```
Comment (gregglind) [inline test/recipes/test-ios-promo.js]: All of this is gross.  IT was gross the first time!  Good catches.

File: test/test-built/test-built-exports.js [hunk 8]
```diff
@@ -42,8 +42,8 @@ describe("built file exports", function () {
     }
   });
   describe("recipe list", function () {
-    it("has 2 recipes.", () => {
-      expect(heartbeat.recipes.length).to.equal(2);
+    it("has 3 recipes.", () => {
```
Comment (gregglind) [inline test/test-built/test-built-exports.js]: file ok

**Comment by gregglind** [comment]
R+.  

Intent:  Remove the lowest performing branch.  Deploy to 10% worldwide.

**Comment by gregglind** [comment]
Oh, new thing!  R-

We have to check for version 40!  

```
state.fxVersion
```

File: src/recipes/ios-promo/index.js (hunk 4 above)
Comment (rayborn) [inline src/recipes/ios-promo/index.js]: That was our solution.  I'll comment it instead.  Okay?

Comment (gregglind) [inline src/recipes/ios-promo/index.js]: Some tests use it.  I would leave it alone.

File: src/recipes/ios-promo/config.js (hunk 3 above)
Comment (rayborn) [inline src/recipes/ios-promo/config.js]: Resolved, leaving as discrete locales to keep in line with previous tests/configs

File: src/recipes/ios-promo/index.js (hunk 4 above)
Comment (rayborn) [inline src/recipes/ios-promo/index.js]: I had edited the tests accordingly

**Pull request merged**